src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyTool.java

```diff
@@ -624,16 +624,20 @@ public void overrideLicenses( LicenseMap licenseMap, SortedMap<String, MavenProj
 
                 if ( ArrayUtils.isEmpty( licenses ) )
                 {
-
                     // empty license means not fill, skip it
+                    LOG.debug( " Not overriding license for dependency [{}] because no override given", id );
                     continue;
                 }
 
-                licenseMap.removeProject( project );
+                // remove project only removes first occurrence of project from license -> project[] map.
+                List<String> removedFrom = licenseMap.removeProject( project );
+                LOG.info( "Overriding license(s) for dependency [{}] with [{}], overriden license(s): [{}]",
```

Comment:
not good print these information to log as info. This plugin is already much verbose.

Reply:
I'm not sure if I agree - yes, it could be considered verbose yet overriding a license can be a major issue which is why I agree with digulla's comment in general.

I'd rather print one summary line (for example, "Overriding license(s) for 123 dependencies. Enable debug log for details.") iff a license has been overridden instead of silently pushing the info down to LOG.debug entirely.

What do you think?

Reply:
I agree with @lama0206

Reply:
After testing with multi module projects with long override list I think that log in debug level will be enough.